lib/instrumentation/modules/memcached.js

```diff
@@ -32,6 +50,8 @@ module.exports = function (memcached, agent, { version, enabled }) {
             arguments[0] = function queryCompiler () {
               return query
             }
+            // Hack to ensure the span is propagated correctly
+            agent._instrumentation.activeSpan = span
```

Comment:
@watson  except for this change, I am satisfied about other changes. 

Without this change, activeSpan would always point to the previous span. Had to update it to add the destination info with correct span. Let me know if i am missing something here or any other way we can trigger the update.